pytorch_lightning/trainer/trainer.py

```diff
@@ -1396,3 +1397,11 @@ def _on_exception(self):
         # save a checkpoint for fault tolerant training. we don't use `log_dir` to minimize the chances of failure.
         file_path = os.path.join(self.default_root_dir, ".pl_auto_save.ckpt")
         self.save_checkpoint(file_path)
+
+    @contextmanager
+    def _evaluation_context(self) -> Generator:
```

Comment:
Should this be owned by the loop instead ?

Reply:
@tchaton Are you suggesting that we should move this context to `evaluation_epoch_loop.py`? I'm just not sure where's best to put it.

Reply:
Yes, I think this could be in the scope of the loops more than the trainer.

Reply:
@tchaton @awaelchli - does this mean the FitLoop needs to override `run` ? if so, that introduces a risk that the fit loop override doesn't stay in sync with the base Loop `run` definition.

asking since anomaly detection is taking a very similar route as what's provided here: https://github.com/PyTorchLightning/pytorch-lightning/pull/9175